packages/flame/test/components/component_test.dart

```diff
@@ -28,6 +28,91 @@ void main() {
         await expectLater(component.mounted, completes);
       });
 
+      testWithFlameGame(
+          'component removed times out when set before game is ready',
+          (game) async {
+        final component = LifecycleComponent('component');
+        final removed = component.removed;
+        await game.add(component);
+        await game.ready();
+
+        game.remove(component);
+        game.update(0);
+        var hasTimedOut = false;
+        await removed.timeout(
+          const Duration(milliseconds: 1),
+          onTimeout: () => hasTimedOut = true,
+        );
+        expect(hasTimedOut, isTrue);
+      });
+
+      testWithFlameGame(
+          'component removed completes when set after game is ready',
+          (game) async {
+        final component = LifecycleComponent('component');
+        await game.add(component);
+        await game.ready();
+        final removed = component.removed;
+
+        game.remove(component);
+        game.update(0);
+        await expectLater(removed, completes);
+      });
+
+      testWithFlameGame(
+        'component removed completes after changing parent',
+        (game) async {
+          final parent = LifecycleComponent('parent')..addToParent(game);
+          final child = LifecycleComponent('child')..addToParent(parent);
+          await game.ready();
+          final removed = child.removed;
+
+          child.changeParent(game);
+          game.update(0);
+          await expectLater(removed, completes);
+
+          final removedFromParent = child.removed;
+          child.removeFromParent();
+          game.update(0);
+          await expectLater(removedFromParent, completes);
+        },
+      );
+
+      testWithFlameGame('remove parent of child that has removed set',
+          (game) async {
+        final parent = LifecycleComponent('parent')..addToParent(game);
+        final child = LifecycleComponent('child')..addToParent(parent);
+        await game.ready();
+        final removed = child.removed;
+
+        parent.removeFromParent();
+        game.update(0);
+        await expectLater(removed, completes);
+        expect(child.isRemoved, true);
+      });
+
+      testWithFlameGame(
+        'component removed times out when changing parent from a null parent',
```

Comment:
Not sure this is an accurate description.
Changing parent from null to A is the same as adding the component to A. So, after that the component will end up mounted -- which means the `removed` future is still pending, and will complete once the component will get removed.

Reply:
Ah right, this test is useless. I will remove it.